.github/workflows/docker-build.yaml

```diff
@@ -11,50 +11,38 @@ jobs:
   container-build:
     name: operator-build
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        go: ["1.20", "1.21"]
     steps:
-      - uses: actions/setup-go@v2
-        with:
-          go-version: ${{ matrix.go }}
+      - uses: actions/checkout@v4
 
-      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v5
+        with:
+            go-version-file: go.mod
 
       - name: Build operator container image
         run: make build
 
   bundle-build:
     name: bundle-build
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        go: ["1.20", "1.21"]
     steps:
-      - uses: actions/setup-go@v2
-        with:
-          go-version: ${{ matrix.go }}
+      - uses: actions/checkout@v4
 
-      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v5
+        with:
+            go-version-file: go.mod
 
       - name: Build operator bundle container image
         run: make bundle-build
 
   catalog-build:
     name: catalog-build
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        go: ["1.20", "1.21"]
     steps:
-      - uses: actions/setup-go@v2
-        with:
-          go-version: ${{ matrix.go }}
+      - uses: actions/checkout@v4
 
-      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v5
+        with:
+            go-version-file: go.mod
 
```

Comment:
Is there a reason for the step order change?

Reply:
Same as https://github.com/red-hat-storage/ocs-client-operator/pull/92#discussion_r1512926724